sycl/include/sycl/builtins.hpp

```diff
@@ -48,18 +48,299 @@ template <class T, int N> marray<T, N> to_marray(vec<T, N> x) {
 namespace __sycl_std = __host_std;
 #endif
 
+#define COMMA ,
+
+#define DEF_BUILTIN_VEC(TYPE)                                                  \
```

Comment:
Oh, one more thing - we should start all internal macro names with a prefix `__SYCL_` to distinguish them from user-level macros.

Reply:
I agree, we should probably follow that naming. In addition, we should probably also undefine them at the end of the header, since they are just intended as generators.

Reply:
The defines are here now https://github.com/intel/llvm/pull/10026